### PR TITLE
feat(gatsby-recipes): Use a queue and render loop for plans

### PR DESCRIPTION
**Pull request by johno**
This should maintain existing functionality, but sets up some of the needed underpinnings for nested resources that share context and will waterfall their children as they suspend. We will also need this functionality when resources are missing data, where they will be able to emit events to the recipes state machine to request more data for a given resource before it's rendered.

Related to #23746, #23747, #23744, #24506

### CODE REVIEW

File: packages/gatsby-recipes/src/renderer/render.test.js [hunk 1]
```diff
@@ -10,6 +12,38 @@ const fixture = (
...
+  expect(result).toMatchInlineSnapshot(`
+    Array [
+      Object {
+        "currentState": "",
+        "describe": "Write red.js",
+        "diff": "- Original  - 0
+    + Modified  + 1
+
+    + red!",
+        "newState": "red!",
```
Comment (KyleAMathews) [inline packages/gatsby-recipes/src/renderer/render.test.js:36]: does this snapshot need updated? It doesn't have the blue/yellow files in it

Comment (johno) [reply packages/gatsby-recipes/src/renderer/render.test.js:36]: The transformer doesn't currently handle nested children, I figured I'd follow along with that as a separate PR into this one (I'll PR that functionality into this branch).

Comment (johno) [reply packages/gatsby-recipes/src/renderer/render.test.js:36]: Resolved with: https://github.com/johno/gatsby/pull/2

**Comment by johno** [comment]
These changes have been pulled into #24595 so going to go ahead and close this.

**Pull request closed without merging**